mesa/space.py

```diff
@@ -551,16 +588,24 @@ def position_agent(
             )
 
         if x == "random" or y == "random":
-            if len(self.empties) == 0:
+            if len(self._empties) == 0:
                 raise Exception("ERROR: Grid full")
             self.move_to_empty(agent)
         else:
             coords = (x, y)
             self.place_agent(agent, coords)
 
+    def move_agent(self, agent: Agent, pos: Coordinate) -> None:
+        """Move an agent from its current position to a new position."""
+        if self.is_cell_empty(pos):
+            Grid.move_agent(self, agent, pos)
+        else:
+            raise Exception("Cell not empty")
+
     def place_agent(self, agent: Agent, pos: Coordinate) -> None:
+        """Place the agent at the specified location, and set its pos variable."""
         if self.is_cell_empty(pos):
-            super().place_agent(agent, pos)
+            Grid.place_agent(self, agent, pos)
```

Comment:
Why replace the `super()`?

Reply:
I think that `super()` is less explicit and should be reserved for more complex cases, it is also 2-2.5 times slower than calling the method of the superclass

Reply:
https://docs.python.org/3/library/functions.html#super
There is the other use case:
> There are two typical use cases for super. In a class hierarchy with single inheritance, super can be used to refer to parent classes without naming them explicitly, thus making the code more maintainable. This use closely parallels the use of super in other programming languages.

Reply:
I don't buy that much this explaination :D, it seems to me that the mantainability is more or less the same, also in our case I think our inheritance tree is rather static so...I have seen many programmers actually against this use case for super :D